app/src/common/shared/com/igalia/wolvic/utils/ConnectivityReceiver.java

```diff
@@ -21,34 +21,83 @@ public interface Delegate {
         void OnConnectivityChanged(boolean connected);
     }
 
-    private Context mContext;
-    private List<Delegate> mListeners;
+    private static class NetworkWatcher extends LiveData<Boolean> {
+        public final ConnectivityManager connectivityManager;
+        public final List<Delegate> listeners;
+        public ConnectivityManager.NetworkCallback networkCallback;
+        public NetworkRequest networkRequest;
+        private final Context context;
+        public NetworkWatcher(@NonNull Context context){
+            this.context = context;
+            this.listeners = new ArrayList<>();
+            this.connectivityManager = context.getSystemService(ConnectivityManager.class);
+            this.networkCallback = new ConnectivityManager.NetworkCallback(){
+                @Override
+                public void onLost(@NonNull Network network) {
+                    NetworkWatcher.super.postValue(false);
+                }
+
+                @Override
+                public void onCapabilitiesChanged(@NonNull Network network, @NonNull NetworkCapabilities networkCapabilities) {
+                    NetworkWatcher.super.postValue(
+                            networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                    );
+                }
+            };
+            this.networkRequest = new NetworkRequest.Builder()
+                    .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+                    .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+                    .addTransportType(NetworkCapabilities.TRANSPORT_VPN)
+                    .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                    .build();
+        }
+
+        public void onReceive() {
+            listeners.forEach(listener -> listener.OnConnectivityChanged(isNetworkAvailable(context)));
+        }
+
+        @Override
+        protected void onActive() {
+            this.connectivityManager.registerDefaultNetworkCallback(this.networkCallback);
```

Comment:
hmm should we use `registerNetworkCallback()` and pass the `networkRequest` here? Otherwise I don't see the point of the latter

Reply:
Ah, I was using that networkRequest with registerNetworkCallback(), but later I find some issues (the callback was not triggered), so I switched to registerDefaultNetworkCallback() but forgot to delete the networkRequest thing. Will remove the networkRequest here.

Reply:
Done